reporting_design.md

```diff
@@ -0,0 +1,130 @@
+previous stages (i.e. those should be encoded in the stage's output), but may
+allow overriding them.
+
+In essence, an output of a stage is an independent entity, using it should not
+require providing any other input to skt.
+
+Each command has inputs and outputs documented explicitly, so they could be
+used both by skt and other programs, however some parts might be defined
+opaque for use by skt only. Specific command descriptions follow.
+
+Merge
+-----
+
+Produce a source tree with patches applied.
+
+### Inputs
+
+#### Command-line options
+
+* git repository to fetch from
+* git reference from the repository to base patches on
+* git fetch depth
+* what to merge on top (in order of specification),
+  arbitrary combination and number:
+    * git reference from the same repository
+    * URL of a patch mbox (including file:// scheme)
+
+#### Input directory
+
+No files expected
+
+### Outputs
+
+#### Output directory
+
+* "source" directory containing what is required as the input by the
+  "Build" command.
+* "merge.log" - diagnostics output of the merging attempt.
+* "merge.done" - empty file, created only if merge stage succeeded
+
+#### Exit status
+Zero on success, one if "merge.done" is missing, two and greater if other
+failure occurred.
+
+Build
+-----
+
+Produce a kernel tarball.
+
+### Inputs
+
+#### Command-line options
+
+* Extra make options
+* Configuration type: "tinyconfig", "rh-configs" (works only if Makefile
+  supports "rh-configs" target), other configuration-generating target
+  supported by the Makefile in "source".
+* Shell glob, relative to the "source" root, matching a configuration file
+  inside the "source" tree to use. Only valid if "rh-configs"
+  configuration type is used.
+* Kernel configuration file to use as the base. Exclusive with
+  "tinyconfig" and "rh-configs" configuration types.
+
+TODO: Figure out a better arrangement, so values, not options are exclusive.
+TODO: See if we can define allowed configuration types more strictly.
+
+#### Input directory
+
+* "source" directory containing the source of the "kernel", must contain
+  the following (TODO: make sure everything we need is listed):
+    * Makefile
+        * Must support the following targets (TODO: detail all targets):
+            * "mrproper" - cleanup the tree
+            * "tinyconfig" - generate minimum kernel configuration
+            * "kernelrelease" - output kernel version
+            * "targz-pkg" - build kernel tarball
+        * May support the following targets:
+            * "rh-configs" - generate a set of configuration files
+              somewhere in the build directory
+            * other targets generating kernel configuration
+* "merge.done" - empty file
+
+### Outputs
+
+#### Output directory
+
+* "kernel.tar.gz" - the built kernel tarball
+* "kernel.config" - the used kernel configuration
+* "build.log" - diagnostics output of the building stage
+* "build.done" - empty file, created only if build stage succeeded
```

Comment:
I'd suggest having a kernelrelease.txt that contains the kernel release text. That would make it easy to parse.

Reply:
Yeah, I expect those are not all the files we need, just a sketch. That can be a YAML file as well. What do you want a `kernelrelease.txt` for?

Reply:
I thought we need that string from `make kernelrelease` when we make tarballs and such? If not, then ignore that request. ;)

Reply:
I'm not sure, let's see where this takes us and what the code needs :)